tools/run-integration-tests.py

```diff
@@ -0,0 +1,182 @@
+import os
+import subprocess
+import tarfile
+import time
+from datetime import datetime
+from sys import stdout
+
+import click
+import yaml
+
+PLATFORMS = (
+    "azuregeneric",
+    "azurepro",
+    "awsgeneric",
+    "awspro",
+    "gcpgeneric",
+    "gcppro",
+    "vm",
+    "lxd",
+    "upgrade",
+)
+
+SERIES_TO_VERSION = {
+    "xenial": "16.04",
+    "bionic": "18.04",
+    "focal": "20.04",
+    "hirsute": "21.04",
+}
+
+TOKEN_TO_ENVVAR = {
+    "prod": "UACLIENT_BEHAVE_CONTRACT_TOKEN",
+    "staging": "UACLIENT_BEHAVE_CONTRACT_TOKEN_STAGING",
+    "expired": "UACLIENT_BEHAVE_CONTRACT_TOKEN_STAGING_EXPIRED",
+}
+
+
+def is_compatible(platform, series):
+    if platform in ("lxd", "upgrade"):
+        return True
+    elif series not in ("hirsute",):
+        return True
+    return False
+
+
+def build_commands(
+    platform, series, proposed, check_version, token, credentials_path, wip
+):
+    with open(credentials_path) as f:
+        credentials = yaml.safe_load(f)
+    commands = []
+    for p in platform:
+        for s in series:
+            # Not all tests can run in all platforms
+            if is_compatible(p, s):
+                series_version = SERIES_TO_VERSION[s]
+                env = os.environ.copy()
+
+                # Add check_version variable
+                if check_version:
+                    env["UACLIENT_BEHAVE_CHECK_VERSION"] = check_version
```

Comment:
@renanrodrigo Okay I think this is my final request for this script :upside_down_face: 
The version is different across ubuntu releases. e.g. focal version is `27.2.2~20.04.1` and bionic version is `27.2.2~18.04.1` etc.
So if we just set one version at the command line then it won't work across all releases' tests.

It's probably good enough for this script to expect the user to just pass `27.2.2` and auto-append the `~$RELEASE.1`

Reply:
Ooh you're right. Will it be always `~$RELEASE.1` though? Do we ever get `~$RELEASE.2` ?

Reply:
To my knowledge, we rarely have needed to use `~$RELEASE.2` - Thats why I only said "probably good enough" :)
Alternatively, we could add another cli flag like "check-version-patch-num" that defaults to `1` that would get used there.

Reply:
Looking at the history, I will leave this flag for later - if we needed, can be easily added in the future